### PR TITLE
Log to external error service

### LINKED ISSUE #5627: Re-enable Rollbar - error tracking service
(or look into Sentry as alternative)

### PR DESCRIPTION
**Pull request by na9da**
### What this PR does

Fixes #5627 

Merge only after #5454 

* Adds a generic error service interface that can be configured through `configParameters`.
* Adds `RollbarErrorServiceProvider`. (by default this logs all uncaught errors and unhandled promise errors).
* Logs to error service in `terria.raiseErrorToUser`.

### Checklist

-   [x] There are unit tests to verify my changes are correct or unit tests aren't applicable (if so, write quick reason why unit tests don't exist)
-   [x] I've updated CHANGES.md with what I changed.

### CODE REVIEW

File: lib/Models/Terria.ts
Comment (nf-s) [inline lib/Models/Terria.ts]: I think we should call `this.errorService.error(terriaError)` regardless of whether the error is shown to the user

Comment (na9da) [reply lib/Models/Terria.ts]: Have changed this to log to service for every call.

**Comment by na9da** [comment]
Thanks for the review @nf-s, I'll merge into `next` once the upstream branch is merged.

**Review by nf-s — CHANGES_REQUESTED** [review]
Hey @na9da 

I was having a play around to see if we could get more useful stack trace in there.

What do you think of this change?

https://github.com/TerriaJS/terriajs/pull/5701

**Comment by na9da** [comment]
Hi @nf-s 

That looks like a much better stack trace. I'll merge it in!

**Review by nf-s — APPROVED** [review]

**Pull request merged**
